### PR TITLE
Porting utils to py3

### PR DESCRIPTION
**Pull request by dincamihai**
## What does this PR change?
Make spacewalk-utils python3 compatible

Changes requested here: https://github.com/SUSE/salt-board/issues/213

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- No tests: no test coverage

- [x] **DONE**

## Changelogs

- [x] **DONE**

### CODE REVIEW

File: utils/cloneByDate.py [hunk 1]
```diff
@@ -19,16 +19,21 @@
 
 import os
 import sys
+import six
```
Comment (meaksh) [inline utils/cloneByDate.py]: I think this should be like:

```python
from salt.ext import six
```

That way we would reuse the `six` implementation we also provide in Salt. Like in https://github.com/uyuni-project/uyuni/pull/285/commits/c331136658a8c2100f5ff47e0f44127d26555784

File: utils/cloneByDate.py [hunk 2]
```diff
@@ -46,21 +51,27 @@
     from common.rhnConfig import CFG, initCFG
     from satellite_tools.progress_bar import ProgressBar
 
-from depsolver import DepSolver
+from builtins import input
```
Comment (meaksh) [inline utils/cloneByDate.py]: Do we have `builtins` provided for both Python 2 and 3 ?

File: utils/cloneByDate.py [hunk 3]
```diff
@@ -868,7 +879,7 @@ def __login(self):
     def list_channel_labels(self):
         self.auth_check()
         key = "chan_labels"
-        if self.cache.has_key(key):
+        if key in self.cache:
```
Comment (meaksh) [inline utils/cloneByDate.py]: Be careful here, there are some RHN classes that implements `has_key()` method and `2to3` is wrongly suggesting changes to `if x in y`. You need to check whether `salt.cache` is really a dict or some other class instance.

File: utils/spacewalk-sync-setup [hunk 4]
```diff
@@ -606,7 +614,11 @@ def describe_master_template(master_setup_filename):
 
 def ask(msg, password=False):
     msg += ": "
-    return password and getpass.getpass(msg) or raw_input(msg)
+    if six.PY2:
```
Comment (meaksh) [inline utils/spacewalk-sync-setup]: We need to import `six` here, right?

File: utils/cloneByDate.py (hunk 1 above)
Comment (mcalmer) [reply utils/cloneByDate.py]: @meaksh ooops. This is not salt. This should not depend on salt.
Also your linked commit might need to be changed

Comment (dincamihai) [reply utils/cloneByDate.py]: we decided to use `salt.ext.six` since this is available on SUSE Manager machine

File: utils/cloneByDate.py (hunk 2 above)
Comment (dincamihai) [reply utils/cloneByDate.py]: This was a leftover. I have removed the line since it was not needed.

File: utils/cloneByDate.py (hunk 3 above)
Comment (dincamihai) [reply utils/cloneByDate.py]: @meaksh it seems it is a dict defined here https://github.com/uyuni-project/uyuni/blob/11da5a723d305016cc73d6519fe79b45028717bb/utils/cloneByDate.py#L854

File: utils/spacewalk-sync-setup (hunk 4 above)
Comment (dincamihai) [reply utils/spacewalk-sync-setup]: @meaksh fixed https://github.com/uyuni-project/uyuni/blob/1976508754ba3b7732ec5db4d3b9e24ff67879ea/utils/spacewalk-sync-setup#L31

Comment (dincamihai) [reply utils/spacewalk-sync-setup]: @meaksh fixed. https://github.com/uyuni-project/uyuni/blob/46419ad08c94af75f7b1cebf8321f2a8e2048cd1/utils/spacewalk-sync-setup#L31

File: utils/sw-ldap-user-sync [hunk 5]
```diff
@@ -69,7 +69,7 @@ for uid in data['member']:
 
     try:
         (userdn, userdata) = directory.search_s(uid, ldap.SCOPE_SUBTREE, filter, attrs)[0]
-        if users.has_key(userdata["uid"][0]):
+        if userdata["uid"][0] in users:
```
Comment (dincamihai) [inline utils/sw-ldap-user-sync]: this is a dict https://github.com/uyuni-project/uyuni/pull/458/commits/d7a39e2ceb44026ed028f57ecf2389edf9b6fb94#diff-df3a719a76e2c83a19bcfe11ecad9b6eR54

File: utils/spacewalk-manage-channel-lifecycle [hunk 6]
```diff
@@ -159,12 +159,12 @@ def print_channel_tree():
...
     for channel in all_channels:
         if channel.get('parent_label'):
-            if not tree.has_key(channel.get('parent_label')):
+            if channel.get('parent_label') not in tree:
                 tree[channel.get('parent_label')] = []
 
             tree[channel.get('parent_label')].append(channel.get('label'))
         else:
-            if not tree.has_key(channel.get('label')):
+            if channel.get('label') not in tree:
```
Comment (dincamihai) [inline utils/spacewalk-manage-channel-lifecycle]: `tree` is a dict: https://github.com/uyuni-project/uyuni/pull/458/commits/d7a39e2ceb44026ed028f57ecf2389edf9b6fb94#diff-5ca5b85da97abc2218f77b86813a4a7eR157

File: utils/spacewalk-clone-by-date [hunk 7]
```diff
@@ -106,7 +106,7 @@ def merge_config(options):
                  "security_only", "use_update_date", "no_errata_sync",
                  "errata", 'dry_run']
     for key in overwrite:
-        if config.has_key(key) and not getattr(options, key):
+        if key in config and not getattr(options, key):
```
Comment (dincamihai) [inline utils/spacewalk-clone-by-date]: `config` is a dict: https://github.com/uyuni-project/uyuni/pull/458/commits/d7a39e2ceb44026ed028f57ecf2389edf9b6fb94#diff-12ad85f09c55ba2f92e47253e811ab68R98

File: utils/cloneByDate.py (hunk 3 above)
Comment (dincamihai) [reply utils/cloneByDate.py]: @meaksh i've check the other `has_key` fixes and the objects are dictionaries. Resolving.

**Comment by dincamihai** [comment]
@mcalmer Pablo finished reviewing. Do you want to take a look before I merge?

**Comment by mcalmer** [comment]
gitarro rerun java_pgsql_tests !!!

**Comment by dincamihai** [comment]
gitarro rerun java_pgsql_tests !!!

**Comment by dincamihai** [comment]
gitaro rerun java_pgsql_tests !!!

**Comment by dincamihai** [comment]
gitarro rerun java_pgsql_tests !!!

**Comment by dincamihai** [comment]
gitarro rerun java_pgsql_tests !!!

**Pull request merged**